core/src/test/java/fj/data/IOFunctionsTest.java

```diff
@@ -56,9 +57,55 @@ public void close() {
       bracketed.run();
       fail("Exception expected");
     } catch (IllegalArgumentException e) {
-      Assert.assertThat(e.getMessage(), Is.is("OoO"));
+      Assert.assertThat(e.getMessage(), is("OoO"));
     }
-    Assert.assertThat(closed.get(), Is.is(true));
+    Assert.assertThat(closed.get(), is(true));
+  }
+
+  @Test
+  public void testTraverseIO() throws IOException {
+    String[] as = {"foo1", "bar2", "foobar3"};
+    Stream<String> stream = Stream.arrayStream(as);
+    ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    PrintStream originalOut = System.out;
+    System.setOut(new PrintStream(outContent));
+    stream.traverseIO(IOFunctions::stdoutPrint).run();
+    System.setOut(originalOut);
+    assertThat(outContent.toString(), is("foobar3bar2foo1"));
+  }
+
+  @Test
+  public void testSequenceWhile() throws IOException {
+    BufferedReader r = new BufferedReader(new StringReader("foo1\nbar2\nfoobar3"));
+    Stream<IO<String>> s1 = Stream.repeat(() -> r.readLine());
+    IO<Stream<String>> io = sequenceWhile(s1, s -> !s.equals("foobar3"));
+    ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    PrintStream originalOut = System.out;
+    System.setOut(new PrintStream(outContent));
+    runSafe(io).foreach(s -> runSafe(stdoutPrint(s)));
```

Comment:
the test is named `testSequenceWhile` but `foreach` is what is actually tested?

Reply:
there is sequenceWhile above. I could split to two tests ...